### PR TITLE
[CHK-1056] Pgs token

### PR DESCRIPTION
**Pull request by stratoivandiluccio**
This PR implements token authentication between FE and BE comunication

List of Changes
Main FE routes modified to get the token from the url and sending it through API calls

Motivation and Context
Check transaction validity

How Has This Been Tested?
Tested Manually simulating the api calls and inspecting the request

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,6 @@
     "start": "parcel src/index.html",
     "dev": "parcel --log-level verbose src/index.html",
     "generate": "npm-run-all -s generate:*",
-    "generate:pgs-be": "shx rm -rf src/generated/pgs && shx mkdir -p src/generated/pgs && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/c39cea0b3ac4d992f4bda0026365b5a0eac96802/src/core/api/payment_transactions_gateway_api/external/v1/_openapi.json.tpl --no-strict --out-dir src/generated/pgs",
```
Comment (Fernando-Granato) [inline package.json:17]: Are you sure you want to eliminate this autogeneration ?

Comment (stratoivandiluccio) [reply package.json:17]: I think this is a temporary solution, suggested by @ciuffagianluca. I suppose that the removed line needs to be restored, but not sure when

Comment (ciuffagianluca) [reply package.json:17]: Hi Fernando,
this autogeneration was completely overwritten by the follower generate (you cannot see in this code portion) so it is completely unuseful. Anyway, we need to update infra json openapi spec and then we can refers it here (coming soon)

**Review by Fernando-Granato — APPROVED** [review]

File: api-spec/pgs_api.yaml [hunk 2]
```diff
@@ -20,6 +20,8 @@ paths:
           required: true
           description: PGS-generated GUID of the request to get
           example: 77e1c83b-7bb0-437b-bc50-a7a58e5660ac
+      security:
```
Comment (infantesimone) [inline api-spec/pgs_api.yaml]: since these spec will no longer be used, it is better to remove the file

Comment (stratoivandiluccio) [reply api-spec/pgs_api.yaml]: file deleted, but please note that the remote api spec specified in the package.json file are actually not available:
https://raw.githubusercontent.com/pagopa/pagopa-infra/jwt-token-auth-pgs/src/core/api/payment_transactions_gateway_api/pgs-fe/v1/_openapi.json.tpl

Comment (ciuffagianluca) [reply api-spec/pgs_api.yaml]: change the refs in this way https://raw.githubusercontent.com/pagopa/pagopa-infra/main/src/core/api/payment_transactions_gateway_api/pgs-fe/v1/_openapi.json.tpl

Comment (stratoivandiluccio) [reply api-spec/pgs_api.yaml]: updated

**Review by infantesimone — APPROVED** [review]

**Pull request merged**